exchanges/huobi/huobi_wrapper.go

```diff
@@ -369,6 +369,27 @@ func (h *HUOBI) FetchTradablePairs(a asset.Item) ([]string, error) {
 			}
 		}
 
+		fmtCurrencies, err := currency.NewPairsFromStrings(pairs)
+		if err != nil {
+			return nil, err
+		}
+
+		var configFormatPairs currency.Pairs
+		var b, q currency.Code
+		for f := range fmtCurrencies {
+			b = fmtCurrencies[f].Base.Lower()
+			q = fmtCurrencies[f].Quote.Lower()
+			if format.Uppercase {
+				b = fmtCurrencies[f].Base.Upper()
+				q = fmtCurrencies[f].Quote.Upper()
+			}
+			configFormatPairs = append(configFormatPairs, currency.Pair{
+				Base:      b,
+				Quote:     q,
+				Delimiter: format.Delimiter,
+			})
+		}
+		return configFormatPairs.Strings(), nil
```

Comment:
This and the `asset.Futures` implementation could be simplified in the `symbols` loop:
```
for z := range symbols {
			if symbols[z].ContractStatus == 1 {
				curr, err := currency.NewPairFromString(symbols[z].ContractCode)
				if err != nil {
					return nil, err
				}
				pairs = append(pairs, format.Format(curr))
			}
		}
```

Reply:
Far more elegant solution! Thanks